package/network/utils/umbim/files/lib/netifd/proto/mbim.sh

```diff
@@ -109,39 +127,103 @@ _proto_mbim_setup() {
+
+	echo "mbim[$$]" "Setting up $ifname"
+	eval $(umbim $DBG -n -t $tid -d $device config | sed 's/: /=/g')
+	tid=$((tid + 1))
 
-	echo "mbim[$$]" "Connected, starting DHCP"
 	proto_init_update "$ifname" 1
 	proto_send_update "$interface"
 
-	json_init
-	json_add_string name "${interface}_4"
-	json_add_string ifname "@$interface"
-	json_add_string proto "dhcp"
-	proto_add_dynamic_defaults
-	json_close_object
-	ubus call network add_dynamic "$(json_dump)"
-
-	json_init
-	json_add_string name "${interface}_6"
-	json_add_string ifname "@$interface"
-	json_add_string proto "dhcpv6"
-	json_add_string extendprefix 1
-	proto_add_dynamic_defaults
-	ubus call network add_dynamic "$(json_dump)"
+	[ -z "$dhcp" ] && dhcp=1
+
+	[ "$pdptype" = "ipv4" -o "$pdptype" = "ipv4v6" ] && {
+		if [ -n "$ipv4address" ]; then
+			json_init
+			json_add_string name "${interface}_4"
+			json_add_string ifname "@$interface"
+			json_add_string proto "static"
+			json_add_array ipaddr
+			json_add_string "" "$ipv4address"
+			json_close_array
+			json_add_string gateway "$ipv4gateway"
+			json_add_array dns
+			[ "$peerdns" = 0 ] || json_add_string "" "$ipv4dnsserver"
+			json_close_array
+			proto_add_dynamic_defaults
+			[ -n "$zone" ] && json_add_string zone "$zone"
+			[ -n "$ip4table" ] && json_add_string ip4table "$ip4table"
+			json_close_object
+			ubus call network add_dynamic "$(json_dump)"
+		elif [ "$dhcp" != 0 ]; then
+			echo "mbim[$$]" "Starting DHCP on $ifname"
+			json_init
+			json_add_string name "${interface}_4"
+			json_add_string ifname "@$interface"
+			json_add_string proto "dhcp"
+			proto_add_dynamic_defaults
+			[ -n "$zone" ] && json_add_string zone "$zone"
+			[ -n "$ip4table" ] && json_add_string ip4table "$ip4table"
+			json_close_object
+			ubus call network add_dynamic "$(json_dump)"
+		fi
+	}
+
+	[ "$pdptype" = "ipv6" -o "$pdptype" = "ipv4v6" ] && {
+		if [ -n "$ipv6address" ]; then
+			json_init
+			json_add_string name "${interface}_6"
+			json_add_string ifname "@$interface"
+			json_add_string proto "static"
+			json_add_array ip6addr
+			json_add_string "" "$ipv6address"
+			json_close_array
+			json_add_string ip6gw "$ipv6gateway"
+			json_add_array dns
+			[ "$peerdns" = 0 ] || json_add_string "" "$ipv6dnsserver"
+			json_close_array
+			proto_add_dynamic_defaults
+			[ -n "$zone" ] && json_add_string zone "$zone"
+			[ -n "$ip6table" ] && json_add_string ip6table "$ip6table"
+			json_close_object
+			ubus call network add_dynamic "$(json_dump)"
+		elif [ "$dhcp" != 0 ]; then
+			echo "mbim[$$]" "Starting DHCPv6 on $ifname"
+			json_init
+			json_add_string name "${interface}_6"
+			json_add_string ifname "@$interface"
+			json_add_string proto "dhcpv6"
+			json_add_string extendprefix 1
+			proto_add_dynamic_defaults
+			[ -n "$zone" ] && json_add_string zone "$zone"
+			[ -n "$ip6table" ] && json_add_string ip6table "$ip6table"
+			json_close_object
+			ubus call network add_dynamic "$(json_dump)"
+		fi
```

Comment:
I don't know how to change on this code to make IPv6 work.
but the code bellow works
```
        [ -n "$ip_6" ] && {
                echo "mbim[$$]" "Configure IPv6 on $ifname"
                ip=${ip_6%%/*}
                mask=${ip_6##*/}
                gateway=$(awk '$1=="ipv6gateway:" {print $2}' "$config")
                mtu=$(awk '$1=="ipv6mtu:" {print $2}' "$config")
                [ "$mtu" ] && ip -6 link set "$ifname" mtu "$mtu"

                proto_add_ipv6_address "$ip" "128"
                proto_add_ipv6_prefix "$ip_6"
                proto_add_ipv6_route "$gateway" "128"
                [ "$defaultroute" = 0 ] || proto_add_ipv6_route "::" 0 "$gateway" "$metric" "" "$ip_6"
                [ "$peerdns" = 0 ] || {
                        dns_servers=$(awk '$1=="ipv6dnsserver:" {printf "%s ",$2}' "$config")
                        for dns in $dns_servers; do
                                proto_add_dns_server "$dns"
                        done
                }   
        }  
```

Reply:
ping @Leo-PL @sch-m @hyc

Reply:
@ptpt52 I'm not sure what are you trying to accomplish. IPv6 worked for me both with static and DHCP configuration. There is no auto-switch between them by design.